chart/templates/deployment.yaml

```diff
@@ -187,7 +187,8 @@ spec:
         - mountPath: /var/log/auditlog
           name: audit-log
 {{- end }}
-{{- if gt (int .Values.auditLog.level) 0 }}
+{{- if eq .Values.auditLog.destination "sidecar" }}
+  {{- if gt (int .Values.auditLog.level) 0 }}
```

Comment:
The docs say: 
> Enabling the API Audit Log with the Helm chart install will create a rancher-audit-log sidecar container in the Rancher pod. This container will stream the log to standard output (stdout). You can view the log as you would any container log.

So why do we need the extra destination param ? Shouldn't it just default to the sidecar, that or the docs need to change. 
Also concerned that anyone with it already one would be broken when they upgrade. If we do go this route we should release note that breaking change.

Reply:
See the explanation in the issue https://github.com/rancher/rancher/issues/26192 cc @dkeightley

Reply:
Yea I read that before my comment and I don't understand what is happening. Specifically this part: 
```
Other details that may be helpful: Logging occurs to both the file and sidecar container - these [should be mutually exclusive](https://rancher.com/docs/rancher/v2.x/en/installation/options/chart-options/#api-audit-log)

A further condition [added to the Helm chart](https://github.com/rancher/rancher/blob/master/chart/templates/deployment.yaml#L146) to populate the sidecar container only when .Values.auditLog.destination equals sidecar.
```

@dkeightley what do you mean by mutually exclusive ? Also why would this change not qualify as a breaking change ? Wouldn't anyone who has audit logging turned on need to change their chart params to keep it that way.

Reply:
Sorry for the delay here. In effect mutually exclusive refers to one or the other - not both.

According to expectation and [our documentation](https://rancher.com/docs/rancher/v2.6/en/installation/install-rancher-on-k8s/chart-options/#api-audit-log) the templating logic doesn't produce the intended outcome, this change is to prevent logging twice (to both sidecar and to hostPath).

For eg:
> Set the auditLog.destination to hostPath to forward logs to volume shared with the host system instead of streaming to a sidecar container.

Reply:
Just to clarify, there is no change needed by users to have the intended outcome, sidecar/hostPath will _only_ log to the chosen destination. The added condition only adjusts the templating to reflect this.